docs/concepts/gas.md

```diff
@@ -0,0 +1,200 @@
+    The amount of gas used by a transaction depends on the transaction itself and is fixed at genesis (ie. it requires 1 unit of gas to store 1 byte of information and it requires 3 units of gas to compute 2+2).  The actual *cost of this gas*, it's price in NEAR tokens is calculated dynamically and on an ongoing basis based on system load.
+
+2. **when an account is charged rent** to remain on the network  \
+    From the NEAR whitepaper, 
+    
+    *"Storage is a long-term scarce asset which is priced on a rental basis.  This means that each account or contract will be charged per byte per block for its storage use deducted automatically from the balance on that account. The storage price is fixed and is only subject to change as a major governance decision."*
+
+    *"If an account has data stored but does not have the resources to pay its storage bill for a given block, that account is deallocated, meaning that its storage will be released back to the chain.  Should the account desire to recover their storage, it will need to reallocate the storage by paying its rent and presenting back the data from the account. The previous state is recoverable by examining the prior history of the chain or third-party backups."*
+
+    It also costs more to control accounts with short names.  At time of writing, all account names must be 2 characters or more and are charged rent up to 10 characters after which rent on account names is zero. At 10 characters long, an account named "abcdefghij" is charged rent some amount `R` (measured in units of gas).  9 characters is `3 x R`, 8 characters is `3 x 3 x R` and so on down to 2 characters which is the most expensive at `3^8 x R` or about `6,500 x R`.  This rent for short names is intended to prevent name squatting.  Refer to the [Key Concept: Account](/docs/concepts/account) page for more detail about accounts.
+
+Thus, accounts have their balance in NEAR tokens and gas is bought on the fly when a transaction gets processed or when the account pays storage rent.
+
+Note that the minimum amount of gas for any transaction is 5000 yoctoNEAR (10^-24 NEAR) at time of writing.
+
+## An Example Scenario
+
+Account has its balance in NEAR tokens
+
+A user decides to send a transaction to the network.  She decides how much gas at most she is willing to spend for the transaction and sets a gas limit for processing the transaction.  Some transactions require an almost constant amount of gas but function calls are hard to predict so a little extra never hurts since whatever remains unspent will be returned.
+
+The transaction reaches the network and, while being processed, it automatically "buys" enough gas to finish processing at the current gas prices.
+
+Once the transaction is processed, the related account is only "charged" for the gas that was used.  Any leftover gas is converted back to NEAR tokens on the account.
+
+
+## What about Prepaid Gas?
```

Comment:
AFAIK the is no way to buy gas and accumulate it. As of onboarding, we use FunctionCall-limited access key which allows to spent a fixed amount of tokens for a specific method call

Reply:
we have a few published statements about this idea of prepaid gas -- i was using this section to try to explain that what it actually means is that a developer funds an account that new users will draw from while onboarding.  once they switch to paid accounts, users would fund themselves.